src/visual.ts

```diff
@@ -370,27 +535,35 @@ module powerbi.extensibility.visual {
                     return <string>colorScale(d.value);
                 });
 
-                heatMap.append(TableHeatMap.HtmlObjTitle).text((d: TableHeatMapDataPoint) => {
-                    if (d.valueStr !== undefined) {
-                        return d.categoryX + ": " + d.valueStr;
-                    }
-                    else {
-                        return d.categoryX + ": " + d.value;
-                    }
+                this.tooltipServiceWrapper.addTooltip(heatMap, (tooltipEvent: TooltipEventArgs<TooltipEnabledDataPoint>) => {
+                    return tooltipEvent.data.tooltipInfo;
                 });
 
                 // legend
+                let legendDataValues = [0].concat(colorScale.quantiles());
+                let legendData = legendDataValues.map((value, index) => {
+                    return {
+                        value: value,
+                        tooltipInfo: [{
+                            displayName: `Min value`,
+                            value: value.toFixed(0)
```

Comment:
It'd be better to keep value as is. That means we shouldn't truncate numbers. We should keep precision as is.

Reply:
It looks poor in this case. Is it exactly need display 6-10 digits after the coma at legend bar?

Reply:
This value is for tooltip only. Tooltip should show value as is.